bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeper.java

```diff
@@ -627,27 +618,15 @@ public LedgerHandle createLedgerAdv(int ensSize, int writeQuorumSize, int ackQuo
     public LedgerHandle createLedgerAdv(int ensSize, int writeQuorumSize, int ackQuorumSize,
                                         DigestType digestType, byte passwd[], final Map<String, byte[]> customMetadata)
             throws InterruptedException, BKException {
-        SyncCounter counter = new SyncCounter();
-        counter.inc();
+        CompletableFuture<LedgerHandle> counter = new CompletableFuture<>();
+
         /*
          * Calls asynchronous version
          */
         asyncCreateLedgerAdv(ensSize, writeQuorumSize, ackQuorumSize, digestType, passwd,
                              new SyncCreateCallback(), counter, customMetadata);
 
-        /*
-         * Wait
-         */
-        counter.block(0);
-        if (counter.getrc() != BKException.Code.OK) {
-            LOG.error("Error while creating ledger : {}", counter.getrc());
-            throw BKException.create(counter.getrc());
-        } else if (counter.getLh() == null) {
```

Comment:
@eolivelli in theory, there should be no null ledgerhandle returned with status code == OK. the validation was there to prevent unexpected condition (e.g. bugs). if possible, I'd suggest not changing the behavior when refactoring this piece of code. does that make sense to you?